lib/combine_pdf/pdf_public.rb

```diff
@@ -403,7 +403,11 @@ def number_pages(options = {})
 
       (opt[:page_range] ? pages[opt[:page_range]] : pages).each do |page|
         # Get page dimensions
-        mediabox = page[:CropBox] || page[:MediaBox] || [0, 0, 595.3, 841.9]
+        mediabox = page[:CropBox] || page[:MediaBox]
+        if mediabox.nil?
+          mediabox = [0, 0, 595.3, 841.9]
```

Comment:
I tested this and it results in an incorrect page size for those 2 PDF's that have issues:

<img width="1229" alt="Screenshot 2024-12-04 at 11 26 32" src="https://github.com/user-attachments/assets/cab2ee22-3541-4ebb-baeb-d3cdc1c0a2ee">

[test w corrupted PDFs.pdf](https://github.com/user-attachments/files/18006460/test.w.corrupted.PDFs.pdf)

Best to dig a bit deeper on how it obtains the metadata for the page size and why it can't read it (while other systems can).

Reply:
But that is the size of the pages of those files. There is only one page where the metadata keys are [:Resources, :Contents, :Parent, :Type], so no mediabox or cropbox, which led to the error. You mean we need to change the size of the pages to the same of the ones we generate?

Some logs when using the corrupted files:

keys [:Type, :Resources, :MediaBox, :Contents, :StructParents, :Parent]
mediabox [0, 0, 595.91998, 842.88]
numbering pages
keys [:Resources, :Contents, :Parent, :Type]
mediabox 

```
        mediabox = page[:CropBox] || page[:MediaBox]
        puts "keys #{page.keys}"
        puts "mediabox #{mediabox}"
        # [:Resources, :Contents, :Parent, :Type]
        if mediabox.nil?
          mediabox = [0, 0, 495.3, 841.9]
          page.mediabox = mediabox
        end
```

Reply:
Not exactly sure if it's the size of those pages or that they should be resized by the library, but the outcome should be that they are indeed all the same size. Try to see how it's done for the PDF's you repaired and mimic the same process/behavior?